controllers/csi/driver/server.go

```diff
@@ -335,23 +306,30 @@ func (svr *CSIDriverServer) loadVolumeInfo(volumeID string) (*metadata.Volume, e
 	if volume == nil {
 		return &metadata.Volume{}, nil
 	}
-	svr.log.Info("loaded volume info", "id", volume.VolumeID, "pod name", volume.PodName, "version", volume.Version, "dynakube", volume.TenantUUID)
+	log.Info("loaded volume info", "id", volume.VolumeID, "pod name", volume.PodName, "version", volume.Version, "dynakube", volume.TenantUUID)
 	return volume, nil
 }
 
-func logGRPC(log logr.Logger) grpc.UnaryServerInterceptor {
+func logGRPC() grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		if info.FullMethod == "/csi.v1.Identity/Probe" {
+		if info.FullMethod == "/csi.v1.Identity/Probe" || info.FullMethod == "/csi.v1.Node/NodeGetCapabilities" {
 			return handler(ctx, req)
 		}
-
-		log.Info("GRPC call", "method", info.FullMethod, "request", req)
-
+		methodName := ""
+		if info.FullMethod == "/csi.v1.Node/NodePublishVolume" {
+			req := req.(csi.NodePublishVolumeRequest)
+			methodName = "NodePublishVolume"
+			log.Info("GRPC call", "method", info.FullMethod, "volume-id", req.VolumeId)
+		} else if info.FullMethod == "/csi.v1.Node/NodeUnpublishVolume" {
+			req := req.(csi.NodeUnpublishVolumeRequest)
+			methodName = "NodeUnpublishVolume"
+			log.Info("GRPC call", "method", info.FullMethod, "volume-id", req.VolumeId)
+		}
 		resp, err := handler(ctx, req)
 		if err != nil {
-			log.Error(err, "GRPC call failed")
+			log.Error(err, fmt.Sprintf("%s GRPC call failed", methodName))
 		} else {
-			log.Info("GRPC call successful", "response", resp)
+			log.Info(fmt.Sprintf("%s GRPC call successful", methodName))
```

Comment:
is this log useful?
would removing the info log before the grpc call and just log error/success reduce log spam?

Reply:
there shouldn't be much spam anymore, but yeah I can't come up with a valid use for this :D